src/main/java/soramitsu/irohautils/balancer/routes/IrohaBalancerRoute.java

```diff
@@ -0,0 +1,98 @@
+    public static final String RABBITMQ_BALANCE_TO_LIST_TORII = "rabbitmq:iroha-balancer?queue=list-torii&routingKey=list-torii" + AMQP_COMMON;
+
+
+    private final IrohaService irohaService;
+    private String[] toriiUris;
+    private String[] listToriiUris;
+
+    @Autowired
+    private CamelContext camelContext;
+
+    public IrohaBalancerRoute(IrohaService irohaService) {
+        this.irohaService = irohaService;
+        toriiUris = irohaService.getIrohaPeers().values()
+                .stream()
+                .map(irohaAPI -> irohaAPI.getUri().toString() + "/iroha.protocol.CommandService_v1?method=torii&synchronous=false")
+                .toArray(String[]::new);
+        listToriiUris = irohaService.getIrohaPeers().values()
+                .stream()
+                .map(irohaAPI -> irohaAPI.getUri().toString() + "/iroha.protocol.CommandService_v1?method=listTorii&synchronous=false")
+                .toArray(String[]::new);
+
+    }
+    private static class ByteArrayList extends ArrayList<byte[]> {}
+
+    @Override
+    public void configure() throws Exception {
+
+        for (String uriString : toriiUris) {
+            log.info("Interceptor for: {}", uriString);
+            URI uri = new URI(uriString);
+            String counterName = uri.getHost() + "__" + uri.getPort();
+            interceptSendToEndpoint(uriString)
+                    .to("metrics:counter:" + counterName + "?increment=1");
+        }
+
+
+        from(RABBITMQ_BALANCE_TO_TORII).routeId("BalanceToTorii")
+                .process(exchange -> {
+                    log.debug("Received transaction to torii");
+                    byte[] body = (byte[]) exchange.getIn().getBody();
+                    TransactionOuterClass.Transaction transaction = TransactionOuterClass.Transaction.parseFrom(body);
+                    exchange.getMessage().setBody(transaction);
+                })
+                .loadBalance()
+                .failover(irohaService.getIrohaPeers().size(), false, true, false, Throwable.class)
+                .to(toriiUris)
+                .end()
+                .to("mock:toriiUris");
+
+        from(RABBITMQ_BALANCE_TO_LIST_TORII).routeId("BalanceToListTorii")
+                .unmarshal().json(JsonLibrary.Jackson, ByteArrayList.class)
+                .process(exchange -> {
+                    ByteArrayList body = exchange.getIn().getBody(ByteArrayList.class);
+                    log.debug("Receive transactions to listTorii, size: {}", body.size());
+                    List<TransactionOuterClass.Transaction> transactions =
+                            body.stream().map(t -> {
+                                try {
+                                    return TransactionOuterClass.Transaction.parseFrom(t);
+                                } catch (InvalidProtocolBufferException e) {
+                                    log.error("Error parsing transaction");
```

Comment:
Why are we logging the same error twice? What is the benefit?

Reply:
if you want to see details you can enable trace level